command/setup_vault.go

```diff
@@ -0,0 +1,675 @@
+	return nil
+}
+
+func (s *SetupVaultCommand) roleExists() bool {
+	existingRoles, _ := s.vLogical.List("/auth/jwt/role")
+	if existingRoles != nil {
+		return slices.Contains(existingRoles.Data["keys"].([]interface{}), vaultRoleTasks)
+	}
+	return false
+}
+
+func (s *SetupVaultCommand) renderRole() (map[string]any, error) {
+	role := map[string]any{}
+	err := json.Unmarshal(vaultRoleBody, &role)
+	if err != nil {
+		return role, fmt.Errorf("[✘] Default auth config text could not be deserialized: %w", err)
+	}
+
+	role["bound_audiences"] = vaultAud
+	role["token_policies"] = []string{vaultPolicyName}
+
+	return role, nil
+}
+
+func (s *SetupVaultCommand) createRoleForTasks(role map[string]any) error {
+	buf, err := json.Marshal(role)
+	if err != nil {
+		return fmt.Errorf("[✘] Role could not be interpolated with args: %w", err)
+	}
+
+	path := "auth/jwt/role/" + vaultRoleTasks
+
+	_, err = s.vLogical.WriteBytes(path, buf)
+	if err != nil {
+		return fmt.Errorf("[✘] Could not create Vault role: %w", err)
+	}
+
+	s.Ui.Info(fmt.Sprintf("[✔] Created role %q.", vaultRoleTasks))
+	return nil
+}
+
+func (s *SetupVaultCommand) policyExists() bool {
+	existingPolicies, _ := s.vClient.Sys().ListPolicies()
+	return slices.Contains(existingPolicies, vaultPolicyName)
+}
+
+func (s *SetupVaultCommand) renderPolicy() (string, error) {
+	secret, err := s.vLogical.Read("sys/auth/jwt")
+	if err != nil {
+		return "", fmt.Errorf("[✘] Could not retrieve JWT accessor: %w", err)
+	}
+	accessor := secret.Data["accessor"].(string)
+
+	policyTextStr := string(vaultPolicyBody)
+	return strings.ReplaceAll(policyTextStr, "auth_jwt_X", accessor), nil
+}
+
+func (s *SetupVaultCommand) createPolicy(policyText string) error {
+	encoded := base64.StdEncoding.EncodeToString([]byte(policyText))
+	policyBody := fmt.Sprintf(`{"policy": "%s"}`, encoded)
+	buf := []byte(policyBody)
+
+	path := "sys/policies/acl/" + vaultPolicyName
+	_, err := s.vLogical.WriteBytes(path, buf)
+	if err != nil {
+		return fmt.Errorf("[✘] Could not create Vault policy: %w", err)
+	}
+
+	s.Ui.Info(fmt.Sprintf("[✔] Created policy %q.", vaultPolicyName))
+
+	return nil
+}
+
+func (s *SetupVaultCommand) authMethodExists() bool {
+	existingConf, _ := s.vLogical.Read("/auth/jwt/config")
+	return existingConf != nil
+}
+
+func (s *SetupVaultCommand) renderAuthMethod() (map[string]any, error) {
+	authConfig := map[string]any{}
+	err := json.Unmarshal(vaultAuthConfigBody, &authConfig)
+	if err != nil {
+		return nil, fmt.Errorf("default auth config text could not be deserialized: %v", err)
+	}
+
+	authConfig["jwks_url"] = s.jwksURL
+	authConfig["default_role"] = vaultRoleTasks
+
+	return authConfig, nil
+}
+
+func (s *SetupVaultCommand) createAuthMethod(authConfig map[string]any) error {
+	buf, err := json.Marshal(authConfig)
+	if err != nil {
+		return fmt.Errorf("auth method could not be interpolated with args: %w", err)
+	}
+	_, err = s.vLogical.WriteBytes("auth/jwt/config", buf)
```

Comment:
We should probably use a non-default name for this to avoid a conflict. 

I uses `jwt-nomad` in the TF module:
https://github.com/hashicorp-modules/terraform-vault-nomad-setup/blob/main/variables.tf#L16